### PR TITLE
Javascript javascript3 week1-Giphy App, weatherApp,HappyProject

### PR DESCRIPTION
**Pull request by RemyaManoharan**

### CODE REVIEW

File: javascript/javascript3/week1/HappyProject/script.js [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+const sad = document.getElementById("sad");
+const neutral = document.getElementById("neutral");
+const Resultdiv = document.getElementsByClassName("fade show");
+
+const getUserAnalysis = async (string) => {
+  const url = `https://twinword-sentiment-analysis.p.rapidapi.com/analyze/?text=${string}`;
+  const options = {
+    method: "GET",
+    headers: {
+      "X-RapidAPI-Key": "e0c2cb79bfmsh74450775b975ec4p1c874bjsn3cbd7fc871e3",
```
Comment (chezzoba) [inline javascript/javascript3/week1/HappyProject/script.js]: You should not have the API key in plain text. Refer to my example code.

File: javascript/javascript3/week1/HappyProject/script.js [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+  if (result) {
+    const { type, score } = result;
+    if (type === "positive") {
+      happy.style.display = "block";
+      sad.style.display = "none";
+      neutral.style.display = "none";
+      Resultdiv[0].classList.add("alert-success");
+      Resultdiv[0].textContent = `Hey there ${name}. You are clearly happy! :)`;
+    } else if (type === "negative") {
+      happy.style.display = "none";
```
Comment (chezzoba) [inline javascript/javascript3/week1/HappyProject/script.js]: Here's a shortened version of this code:

```javascript
const displaySentiment = (type, name) => {
  const messages = {
    positive: `Hey there ${name}. You are clearly happy! :)`,
    negative: `I'm sorry ${name}. You are clearly sad! :(`,
    neutral: `Hello ${name}. You are so-so :|`
  };
  const faces = {
    positive: happy,
    negative: sad,
    neutral: neutral
  };
  const classes = {
    positive: "alert-success",
    negative: "alert-primary",
    neutral: "alert-info"
  };

  Object.keys(messages).forEach(key => {
    const display = key === type ? "block" : "none";
    faces[key].style.display = display;
    Resultdiv[0].classList.toggle(classes[key], key === type);
    Resultdiv[0].textContent = messages[type];
  });
};

const analysAndDisplay = (result, name) => {
  if (result) {
    displaySentiment(result.type, name);
  } else {
    Resultdiv[0].classList.add("alert-danger");
    Resultdiv[0].textContent = `Hello ${name}. We couldn't determine your sentiment. Please try again.`;
  }
};
```

File: javascript/javascript3/week1/coolAPI/cool.js [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+  } catch (err) {
+    console.log(err);
+  }
+};
+getAuthors();
+// the data we got is here a object havings keys 'entries','size', 'links'
+
+const displayAuthors = async (books) => {
+  const titles = books.map((book) => {
+    return book.title;
```
Comment (chezzoba) [inline javascript/javascript3/week1/coolAPI/cool.js]:
```javascript
const titles = books.map( ( { title } ) => title);
```

File: javascript/javascript3/week1/coolAPI/cool.js [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+  });
+  titles.forEach((title) => {
+    const litag = document.createElement("li");
+    litag.innerHTML = title;
```
Comment (chezzoba) [inline javascript/javascript3/week1/coolAPI/cool.js:24]: Use innerText not innerHTML

File: javascript/javascript3/week1/giphyApp/giphy.js [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+
+btnTag.addEventListener("click", async (e) => {
+  e.preventDefault();
+  const searchTerm = document.getElementById("search-id").value;
+  const resultNumber = document.getElementById("count").value || 1;
+  searchGiphy(searchTerm, resultNumber);
+});
+const searchGiphy = async (term, count) => {
+  try {
+    const apiKey = "mSA30PH3U1iP124OtC2BNkS9NJyiTfRC";
```
Comment (chezzoba) [inline javascript/javascript3/week1/giphyApp/giphy.js]: API key in plain text

File: javascript/javascript3/week1/weatherApp/weather.js [hunk 6]
```diff
@@ -0,0 +1,148 @@
+const cityTag = document.getElementById("city");
+const btnTag = document.getElementById("submit");
+const locationBtnTag = document.getElementById("search");
+const messageTag = document.getElementById("message");
+const dispalyInfo = document.getElementById("weatherinfo-wrapper");
+
+const saveLocationToLocalStorage = (location) => {
+  localStorage.setItem("savedLocation", JSON.stringify(location));
```
Comment (chezzoba) [inline javascript/javascript3/week1/weatherApp/weather.js:10]: Cool way to cache the location.

File: javascript/javascript3/week1/weatherApp/weather.js [hunk 7]
```diff
@@ -0,0 +1,148 @@
...
+  }
+});
+
+const getCurrentLocation = async () => {
+  try {
+    const position = await new Promise((resolve, reject) => {
+      navigator.geolocation.getCurrentPosition(
+        (position) => {
+          resolve(position);
+        },
```
Comment (chezzoba) [inline javascript/javascript3/week1/weatherApp/weather.js:57]: Shorten it:

```javascript
const position = await navigator.geolocation.getCurrentPosition()
```

File: javascript/javascript3/week1/weatherApp/weather.js [hunk 8]
```diff
@@ -0,0 +1,148 @@
...
+    console.log(error);
+    throw new Error("Failed to get current location");
+  }
+};
+
+// get the lat,lon from city name
+const getCityLocation = async (cityname) => {
+  try {
+    const city = await fetch(
+      `http://api.openweathermap.org/geo/1.0/direct?q=${cityname}&limit=1&appid=a97358faa2b07b8bfa9e60a679c71b47`
```
Comment (chezzoba) [inline javascript/javascript3/week1/weatherApp/weather.js]: appid in plain text

File: javascript/javascript3/week1/weatherApp/weather.js [hunk 9]
```diff
@@ -0,0 +1,148 @@
...
+if (savedLocation) {
+  getWeatherInfo(savedLocation.lat, savedLocation.lon).then((result) => {
+    renderWeatherInfo(result);
+  });
+}
+function showTime() {
+  const currentdate = new Date();
+  const datetime =
+    currentdate.getDate() +
+    "/" +
```
Comment (chezzoba) [inline javascript/javascript3/week1/weatherApp/weather.js:141]: Use the toLocaleString method to format date into string
```javascript
function showTime() {
  const currentdate = new Date();
  const options = { day: "numeric", month: "numeric", year: "numeric", hour: "numeric", minute: "numeric" };
  return currentdate.toLocaleString(undefined, options);
}
```

File: javascript/javascript3/week1/weatherApp/weather.js [hunk 10]
```diff
@@ -0,0 +1,148 @@
...
+  const sunset = document.getElementById("sunset");
+  const headTag = document.getElementById("date-tag");
+
+  const time = showTime();
+  headTag.innerHTML = time;
+  city1.innerHTML = `${data.name}`;
+  temperature.innerHTML = `${Math.round(data.main.temp)}°C`;
+  icon.src = `http://openweathermap.org/img/wn/${data.weather[0].icon}.png`;
+  speed.innerHTML = `Wind Speed : ${data.wind.speed}m/s`;
+  cloudiness.innerHTML = `Cloudiness : ${data.clouds.all}%`;
```
Comment (chezzoba) [inline javascript/javascript3/week1/weatherApp/weather.js]: Could be rewritten to not use innerHTML:

```javascript
const time = showTime();
headTag.textContent = time;
city1.textContent = data.name;
temperature.textContent = `${Math.round(data.main.temp)}°C`;
icon.src = `http://openweathermap.org/img/wn/${data.weather[0].icon}.png`;
speed.textContent = `Wind Speed : ${data.wind.speed}m/s`;
cloudiness.textContent = `Cloudiness : ${data.clouds.all}%`;
const sunriseTime = new Date(data.sys.sunrise * 1000).toLocaleTimeString("en-US");
sunrise.textContent = `Sunrise : ${sunriseTime}`;
const sunsetTime = new Date(data.sys.sunset * 1000).toLocaleTimeString("en-US");
sunset.textContent = `Sunset : ${sunsetTime}`;
```

**Review by chezzoba — COMMENTED** [review]
Really well done, you are clearly very good at programming. I would recommend not using innerHTML and innerText interchangebly.

**Review by chezzoba — COMMENTED** [review]
Really well done, you are clearly very good at programming. I would recommend not using innerHTML and innerText interchangebly.

File: javascript/javascript3/week1/giphyApp/giphy.js (hunk 5 above)
Comment (RemyaManoharan) [reply javascript/javascript3/week1/giphyApp/giphy.js]: will keep api key in secret env.json file

**Review by chezzoba — APPROVED** [review]
Nicely done, I see no more API keys and innerHTML so it seems like your code is good. Try to see, as you are learning node, if you can write shorter and cleaner code in the future.

**Comment by RemyaManoharan** [comment]
Thank you for the time and feedback.. :)

**Pull request merged**